operator/src/main/java/org/bf2/operator/operands/KafkaCluster.java

```diff
@@ -718,9 +720,18 @@ private Map<String, Object> buildKafkaConfig(ManagedKafka managedKafka, Kafka cu
         config.put("kas.policy.shared-admin.adminclient-listener.protocol", "SSL");
 
         if (cruiseControlEnabled) {
-            config.put("cruise.control.metrics.topic.min.insync.replicas", this.configs.getConfig(managedKafka).cruiseControl.getMetricReporterTopicMinInsyncReplicas());
+            config.put("cruise.control.metrics.topic.min.insync.replicas", instanceConfig.cruiseControl.getMetricReporterTopicMinInsyncReplicas());
         }
 
+        Quantity cpu = new Quantity(kafkaConfigs.getContainerCpu());
+        BigDecimal cpuBytes = Quantity.getAmountInBytes(cpu);
+        // cpuBytes might be less than 1
+        int cpuCores = Math.max(1, cpuBytes.intValue());
+
+        int volumeCount = kafkaConfigs.getVolumeCount();
+        // since the thread number is per data dir, we should consider the volume count in the broker
+        config.put("num.recovery.threads.per.data.dir", cpuCores / volumeCount);
```

Comment:
Looking at the [Kafka docs](https://kafka.apache.org/documentation/#brokerconfigs_num.recovery.threads.per.data.dir), this must indeed be an int, but it looks like only positive ints are valid `[1,...]`. I think there are cases where this calculcation could result with `0`, which sounds like an invalid value on the broker.

E.g. if we have 4 CPU cores, but 5 volumes, then the result here will be `0` (or any time we have more volumes than CPU cores. Is there anything preventing us from such a scenario in the future? I think we should maybe say that this needs to be at least 1. WDYT?

I wonder if it would make sense to not force `cpuCores` to be an int, and not force it to be 1, then allow decimal division here, and then only round the result in the end (and make sure it's at least 1)?

Reply:
Good point! I've updated it with:
1. use `double` for `cpuCores`
2. use Math.max(1, cpuCores/volumes) to avoid 0 result.

 Thanks.